backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/util/LinkHelper.java

```diff
@@ -225,18 +225,26 @@ private static void unsetInlineResource(BaseResource model, Class<?> type) {
      * @return           the parent object, or null if not set
      */
     private static <R extends BaseResource> BaseResource getParent(R model, Class<?> parentType) {
+        Object selectedParent = null;
+
+        // Try to match a method to parentType first, only fallback to a subclass if parentType was
```

Comment:
`Try to match a method to parentType`
needs rephrasing - Try to method whose parent matches parentType?